keps/sig-api-machinery/3157-watch-list/README.md

```diff
@@ -581,6 +582,63 @@ Then on the server side we:
 3. reject the request if waitUntilFreshAndBlock times out, thus forcing informers to retry.
 4. otherwise, construct the final list and send back to a client.
 
+### Replacing standard List request with WatchList mechanism for client-go's List method.
+
+Replacing the underlying implementation of the List method for client-go based clients (like typed or dynamic client) 
+with the WatchList mechanism requires ensuring that the data returned by both the standard List request and 
+the new WatchList mechanism remains identical. The challenge is that WatchList no longer retrieves the entire 
+list from the server at once but only receives individual items, which forces us to "manually" reconstruct 
+the list object on the client side.
+
+To correctly construct the list object on the client side, we need type information such as: Group, Version, and Kind. 
+However, simply reconstructing the list object based on these data is not enough. 
+In the case of a standard List request, the server's response (a versioned list) is processed through a chain of decoders, 
+which can potentially modify the resulting list object. 
+A good example is the WithoutVersionDecoder, which removes the GVK information from the list object.
+
+
+To ensure full compatibility, the server must provide a versioned empty list in a format that the client can decode. 
+Otherwise, the "manually" constructed list object may not be consistent with the transformations applied by the decoders, 
+leading to differences.
+
+The proposed solution is to add a new annotation (`k8s.io/initial-events-embedded-list`) to the object returned 
+in the bookmark event (The bookmark event is sent when the state is synced and marks the end of WatchList stream). 
+This annotation will store an empty, versioned list encoded as a Base64 string. 
+This annotation will be added to the same object/place the `k8s.io/initial-events-end` annotation is added.
+
+When the client receives such a bookmark, it will base64 decode the empty list and pass it to the decoder chain. 
+Only after a successful response from the decoders the list will be populated with data received from subsequent 
+watch events and returned.
```

Comment:
missing mentioning "protobuf" or other binary formats. With JSON-only GVK would be enough because we could always reconstruct that string. But the point is that we want that the actual encoding format is used such that we can faithfully reproduce the list object construction through the decode chain.

Reply:
This comment is critical to understanding why we need something more than GVK.  Please be sure to add it.

Reply:
I added more details to the paragraph that starts with `To ensure full compatibility`.